### PR TITLE
chore: Remove channel syntax

### PR DESCRIPTION
**Pull request by paulbutcher**
Further to #4755 this removes the surface channel syntax (and fixes up everwhere it's used within the library and tests)

### CODE REVIEW

**Comment by magnus-madsen** [comment]
@mlutze Lets both do a scan to ensure that everything looks OK.

File: examples/sending-and-receiving-on-channels.flix [hunk 1]
```diff
@@ -1,22 +1,22 @@
 /// A function that sends every element of a list
-def send(c: Channel[Int32], l: List[Int32]): Unit \ IO =
```
[the remark this answers is not in the document]
Comment (mlutze) [reply examples/sending-and-receiving-on-channels.flix:2]: Also, these don't follow subject-last, but can be fixed later.

File: main/src/ca/uwaterloo/flix/language/phase/Parser.scala [hunk 2]
```diff
@@ -743,8 +739,8 @@ class Parser(val source: Source) extends org.parboiled2.Parser {
     def Primary: Rule1[ParsedAst.Expression] = rule {
       Static | Scope | LetMatch | LetMatchStar | LetRecDef | LetUse | LetImport | IfThenElse | Reify | ReifyBool |
         ReifyType | ReifyPurity | Choose | TypeMatch | Match | LambdaMatch | Try | Lambda | Tuple |
-        RecordOperation | RecordLiteral | Block | RecordSelectLambda | NewChannel |
```
Comment (mlutze) [inline main/src/ca/uwaterloo/flix/language/phase/Parser.scala:746]: Is `PutChannel` already gone?

File: main/test/flix/Test.Exp.Concurrency.Select.flix [hunk 3]
```diff
@@ -2,392 +2,392 @@ namespace Test/Exp/Concurrency/Select {
...
+        case _ <- r13 => select {
+        case _ <- r11 => ()
+        case _ <- r11 => ()
+        } ; Channel.recv(r9) ; Channel.recv(r10) ; Channel.recv(r12) ; let (s42, r42) = Channel.buffered(0);
+        let (s43, r43) = Channel.buffered(0);
+        let (s44, r44) = Channel.buffered(0);
+        let (s45, r45) = Channel.buffered(0);
+        spawn { Channel.send((), s45) ; () } ; spawn { Channel.send((), s44) ; () } ; spawn { Channel.send((), s43) ; () } ; spawn { Channel.send((), s42) ; () } ; select {
+        case _ <- r43 => Channel.recv(r42) ; Channel.recv(r44) ; Channel.recv(r45)
+        case _ <- r42 => Channel.recv(r45) ; Channel.recv(r43) ; Channel.recv(r44) ; ()
```
Comment (magnus-madsen) [inline main/test/flix/Test.Exp.Concurrency.Select.flix:140]: Ah yes, good times. These are the only auto-generated tests in Flix.

File: main/src/ca/uwaterloo/flix/language/phase/Parser.scala (hunk 2 above)
Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/Parser.scala:746]: PutChannel was elsewhere in the grammar (not primary).

File: examples/sending-and-receiving-on-channels.flix [hunk 4]
```diff
@@ -1,22 +1,22 @@
...
+def send(s: Sender[Int32], l: List[Int32]): Unit \ IO =
```
[the remark this answers is not in the document]
Comment (paulbutcher) [reply examples/sending-and-receiving-on-channels.flix:2]: Agreed. I wanted to keep the changes (which are already pretty extensive) to the minimum to get things compiling again. But I'm happy to address this if we have  concrete suggestion for what might be better?

**Comment by paulbutcher** [comment]
The only two substantive comments seem to be around possibly renaming `send` functions? I'm happy to make changes if we have a concrete suggestion for what they should be changed to? Or perhaps we should go ahead and merge this and make those changes under a separate PR?

**Comment by magnus-madsen** [comment]
> I vote we merge this and make the name changes as an immediate followup.
> 
> Name suggestions: `send` -> `sendAll`, `recv` -> `recvN`

Let's do that. I will merge.

**Pull request merged**